config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -67,6 +76,36 @@ public void getConfig(GetConfigRequest request,
     }
   }
 
+  @Override
+  public void getAllConfigs(GetAllConfigsRequest request,
+      StreamObserver<GetAllConfigsResponse> responseObserver) {
+    try {
+      List<ContextSpecificConfig> contextSpecificConfigList = configStore
+          .getAllConfigs(request.getResourceName(), request.getResourceNamespace(), getTenantId());
+      responseObserver.onNext(
+          GetAllConfigsResponse.newBuilder().addAllContextSpecificConfigs(contextSpecificConfigList)
+              .build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      log.error("Get all configs failed for request:{}", request, e);
+      responseObserver.onError(e);
+    }
+  }
+
+  @Override
+  public void deleteConfig(DeleteConfigRequest request,
+      StreamObserver<DeleteConfigResponse> responseObserver) {
+    try {
+      // write an empty config for the specified config resource. This maintains the versioning.
```

Comment:
Hmmm - would we expect a deletion to remove historical versions too? Is there ever a way to delete those?

Reply:
I think the main reason for keeping the historical versions is to be able to retrieve the old versions. Here also, if the client deletes the config either by mistake or for some other reason and wants to retrieve the older version, I think it should be able to do it (assuming we expose the API for retrieving the old versions going forward).
With the current scheme of things, there is no way to delete the old versions. We may want to put a cap on the no. of recent versions maintained, and probably clean up the old versions on the server side.

Reply:
Maintaining is fine for now - once we have a better idea how versions will be used we can revisit the implementation since the API won't change.